server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/remoteDelivery/AddressesArrayToMailAddressListConverter.java

```diff
@@ -40,19 +39,20 @@
         if (addresses == null) {
             return ImmutableList.of();
         }
-        return FluentIterable.from(Arrays.asList(addresses))
-            .transform(address -> toMailAddress(address))
+        return Arrays.asList(addresses)
+            .stream()
+            .map(address -> toMailAddress(address))
             .filter(Optional::isPresent)
```

Comment:
.flatMap(OptionalUtils::toStream) is better IMO

Reply:
And added one more test